homeassistant/components/zeversolar_local/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""Local access to the zeversolar invertor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from zeversolarlocal import api
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import COORDINATOR, DOMAIN, ZEVER_URL
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Local access to the zeversolar invertor from a config entry."""
+
+    zever_url = entry.data[ZEVER_URL]
+
+    async def async_update_data() -> api.SolarData:
+        """Get solar data from the zever solar inverter."""
+        try:
+            # not using a timeout here. Timeout is managed by the
+            # zeversolarlocal package itself as it is a vital part of the
+            # working of the package.
+            _LOGGER.debug("Updating zeversolar data")
```

Comment:
Put the debug log before the try.

Reply:
https://github.com/home-assistant/core/pull/55440/commits/defc252594af4de962857a5844f075c3148dfbb5